File: demo/src/main/java/com/bytedance/scenedemo/animation/TransitionDemo.java
package com.bytedance.scenedemo.animation;

import android.os.Build;
import android.os.Bundle;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.annotation.RequiresApi;
import androidx.transition.AutoTransition;
import androidx.transition.ChangeTransform;
import androidx.transition.Slide;
import androidx.transition.Transition;

import com.bytedance.scene.animation.NavigationTransitionExecutor;
import com.bytedance.scene.group.GroupScene;
import com.bytedance.scene.interfaces.PushOptions;
import com.bytedance.scenedemo.R;
import com.bytedance.scenedemo.utility.ColorUtil;

/**
 * Created by JiangQi on 8/23/18.
 */
@RequiresApi(api = Build.VERSION_CODES.O)
public class TransitionDemo extends GroupScene {

    @NonNull
    @Override
    public ViewGroup onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return (ViewGroup) inflater.inflate(R.layout.layout_transition_0, container, false);
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        getView().findViewById(R.id.btn).setOnClickListener(new View.OnClickListener() {
            @RequiresApi(api = Build.VERSION_CODES.O)
            @Override
            public void onClick(View v) {
                getNavigationScene().push(TransitionDemo2.class, null, new PushOptions.Builder().setAnimation(new Test()).build());
            }
        });
    }

    @Override
    public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        getView().setBackgroundColor(ColorUtil.getMaterialColor(getResources(), 1));
    }

    public static class TransitionDemo2 extends GroupScene {

        @NonNull
        @Override
        public ViewGroup onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
            return (ViewGroup) inflater.inflate(R.layout.layout_transition_1, container, false);
        }

        @Override
        public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
            super.onViewCreated(view, savedInstanceState);
            getView().setBackgroundColor(ColorUtil.getMaterialColor(getResources(), 2));
        }
    }

    class Test extends NavigationTransitionExecutor {
        @Override
        protected Transition getSharedElementTransition() {
            Transition autoTransition = new AutoTransition().addTransition(new ChangeTransform());
            autoTransition.addTarget("wo");
            autoTransition.addTarget("imageview");
            return autoTransition;
        }

        @Override
        protected Transition getOthersTransition() {
            Slide slide = new Slide();
            slide.setSlideEdge(Gravity.BOTTOM);
            return slide;
        }
    }
}

File: demo/src/main/java/com/bytedance/scenedemo/animation/TransitionDemo.kt
package com.bytedance.scenedemo.animation

import android.os.Build
import androidx.annotation.RequiresApi
import com.bytedance.scene.group.GroupScene
import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import com.bytedance.scenedemo.R
import com.bytedance.scene.interfaces.PushOptions
import com.bytedance.scene.animation.NavigationTransitionExecutor
import android.view.Gravity
import android.view.View
import androidx.transition.AutoTransition
import androidx.transition.ChangeTransform
import androidx.transition.Slide
import androidx.transition.Transition
import com.bytedance.scene.ktx.requireNavigationScene
import com.bytedance.scenedemo.utility.ColorUtil

/**
 * Created by JiangQi on 8/23/18.
 */
@RequiresApi(api = Build.VERSION_CODES.O)
class TransitionDemo : GroupScene() {
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup, savedInstanceState: Bundle?): ViewGroup {
        return inflater.inflate(R.layout.layout_transition_0, container, false) as ViewGroup
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        view.findViewById<View>(R.id.btn).setOnClickListener {
            requireNavigationScene().push(
                TransitionDemo2::class.java, null, PushOptions.Builder().setAnimation(
                    Test()
                ).build()
            )
        }
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        getView().setBackgroundColor(ColorUtil.getMaterialColor(resources, 1))
    }

    class TransitionDemo2 : GroupScene() {
        override fun onCreateView(
            inflater: LayoutInflater,
            container: ViewGroup,
            savedInstanceState: Bundle?
        ): ViewGroup {
            return inflater.inflate(R.layout.layout_transition_1, container, false) as ViewGroup
        }

        override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
            super.onViewCreated(view, savedInstanceState)
            getView().setBackgroundColor(ColorUtil.getMaterialColor(resources, 2))
        }
    }

    internal inner class Test : NavigationTransitionExecutor() {
        override fun getSharedElementTransition(): Transition {
            val autoTransition: Transition = AutoTransition().addTransition(ChangeTransform())
            autoTransition.addTarget("wo")
            autoTransition.addTarget("imageview")
            return autoTransition
        }

        override fun getOthersTransition(): Transition {
            val slide = Slide()
            slide.slideEdge = Gravity.BOTTOM
            return slide
        }
    }
}